### PR TITLE
Feature/s7 rostering schedule2.0 js

### LINKED ISSUE #383: Add split openning hours to rostering DAG
**Is your feature request related to a problem? Please describe.**
The work center should be able to have split opening hours, for example, open from 09:00 to 14:00 and from 16:0 to 21:00.

**Describe the solution you'd like**
The input data should be able to declare this situation and the model to take it accordingly

**Describe alternatives you've considered**
None

**Additional context**
None

### LINKED ISSUE #384: Add different openning times and closing times for the work center in the rostering DAG
**Is your feature request related to a problem? Please describe.**
The opening and closing time of the working center does not have to be the same every day. The model should be able to take into account that some days the work center will open later and close earlier, for example.

**Describe the solution you'd like**
To be able to set upo opening and closing times for each day of the week and for the model to take those into account.

**Describe alternatives you've considered**
None

**Additional context**
None

### LINKED ISSUE #385: Have exceptions on openning or closing time on rostering DAG
**Is your feature request related to a problem? Please describe.**
The store could open a given day later or sooner. The same for the closing time. Or it could be that one day is not going to open even if it is not a holiday

**Describe the solution you'd like**
It should be possible to set up an exception for a given day on opening or closing time or even on the aperture the store as a whole and the model should take it into account

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by JaimeSotomayor**
closes #383, closes  #384, closes #385

### CODE REVIEW

File: cornflow-dags/DAG/rostering/core/instance.py [hunk 1]
```diff
@@ -338,18 +372,14 @@ def _hour_to_slot(self, hours) -> int:
...
-
     def _get_employees_contracts(self) -> SuperDict[Tuple[int, int], int]:
         """
         Returns a SuperDict with the week and employee tuple as key
         and the contract id as value
         Contracts are supposed to start on Monday and end on Sunday
         For example: {(36, 1): 10, ...}
         """
-        default_date = get_date_string_from_ts(self._get_end_date())
+        default_date = get_date_string_from_ts(self.dates[-1])
```
Comment (ggsdc) [inline cornflow-dags/DAG/rostering/core/instance.py:382]: I know that dates is sorted, but I would prefer that instead of it being `self.dates[-1]` we keep a `self._get_end_date()` which in turns either it returns the maximum date in `self.dates` to make sure that even if something happens and it is not properly sorted that we get the last date from the horizon.

File: cornflow-dags/DAG/rostering/core/instance.py [hunk 2]
```diff
@@ -662,14 +683,15 @@ def get_incompatible_slots_employee(self) -> TupList:
...
-        return TupList(
+        ts_tuple = TupList(
             [
                 (ts, ts2, e)
                 for (ts, ts2) in self._get_incompatible_slots()
                 for e in self._get_employees("name")
                 if (ts, e) in availability and (ts2, e) in availability
             ]
         )
+        return ts_tuple
```
Comment (ggsdc) [inline cornflow-dags/DAG/rostering/core/instance.py:694]: Why this change?

**Review by ggsdc — CHANGES_REQUESTED** [review]

File: cornflow-dags/DAG/rostering/core/instance.py (hunk 2 above)
Comment (JaimeSotomayor) [reply cornflow-dags/DAG/rostering/core/instance.py:698]: I like it better and helps me when debugging since i have an object to review
